### PR TITLE
Fix outdated Mesos paper URL

### PR DESCRIPTION
**Pull request by martyanov**

### CODE REVIEW

**Comment by martyanov** [comment]
Check CLA.

File: contributors/design-proposals/architecture/architecture.md [hunk 1]
```diff
@@ -170,7 +170,7 @@ Kubernetes supports user-provided schedulers and multiple concurrent cluster sch
 using the shared-state approach pioneered by 
 [Omega](https://research.google.com/pubs/pub41684.html). In addition to the disadvantages of
 pessimistic concurrency described by the Omega paper, 
-[two-level scheduling models](http://mesos.berkeley.edu/mesos_tech_report.pdf) that hide information from the upper-level
+[two-level scheduling models](https://people.eecs.berkeley.edu/~alig/papers/mesos.pdf) that hide information from the upper-level
```
Comment (cblecker) [inline contributors/design-proposals/architecture/architecture.md]: https://amplab.cs.berkeley.edu/wp-content/uploads/2011/06/Mesos-A-Platform-for-Fine-Grained-Resource-Sharing-in-the-Data-Center.pdf looks like the official dated version

Comment (martyanov) [reply contributors/design-proposals/architecture/architecture.md]: I'm ok with both. :)

Comment (martyanov) [reply contributors/design-proposals/architecture/architecture.md]: If it's necessary I could change the link to the one suggested.

Comment (cblecker) [reply contributors/design-proposals/architecture/architecture.md]: Please amend the commit with this URL. Thank you!

**Comment by martyanov** [comment]
Done.

**Pull request merged**